javascript/modules/client/src/internal/client-core.ts

```diff
@@ -26,8 +27,14 @@ export abstract class ClientCore implements IClientCore {
   private _signer: Signer | undefined;
   private _daoFactoryAddress = "";
   private _gasFeeEstimationFactor = 1;
+  private _ipfs: IPFSHTTPClient[] = [];
+  private _ipfsIdx: number = -1;
 
   constructor(context: Context) {
+    if (context.ipfs) {
+      this._ipfs = context.ipfs;
+      this._ipfsIdx = Math.floor(Math.random() * context.ipfs.length);
```

Comment:
If the array had 0 elements, we would have an index of `0`, which doesn't exist.
I'd throw an error if so